File: Difficulty: Basic/Linked List End Insertion/linked-list-end-insertion.cpp
/*
class Node {
  public:
    int data;
    Node* next;

    // Default constructor
    Node() {
        data = 0;
        next = NULL;
    }

    // Parameterised Constructor
    Node(int data) {
        this->data = data;
        this->next = NULL;
    }
};
*/
class Solution {
  public:
    Node *insertAtEnd(Node *head, int x) {
        // Code here
        Node* temp = new Node(x);
        if(head == NULL)
        return temp;
        
        Node* curr = head;
        
        while(curr->next != NULL){
            curr = curr->next;
        }
         curr->next = temp;
         return head;
    }
};

File: Difficulty: Basic/Linked List End Insertion/linked-list-end-insertion.py
'''    
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
'''

class Solution:
    def insertAtEnd(self, head, x):
        n = Node(x)
        if not head: return n
        last = head
        while last.next: last = last.next
        last.next = n
        return head